Alerting: Add delete bulk action in the alert list view

soniaAguilarPeiron opened the pull request:
**What is this feature?**

This PR adds **delete** bulk actions for a particular folder. (_pause/unpause are implemented and disabled as they will be enabled once we have the new api_)
This feature is behind the ff `alertingBulkActionsInUI` enabled by default, and it's only available for `admin` users.

ephemeral instance: https://ephemeral15111821104611soniaa.grafana-dev.net/ 

**Why do we need this feature?**

After importing rules from DMA to GMA , users often need to clean up or manage them in bulk.

**Who is this feature for?**

admin alerting users.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/alerting-squad/issues/1105 

**Special notes for your reviewer:**

**Deleting a folder:**


https://github.com/user-attachments/assets/9fd9de7d-c83e-47c1-afbd-0d48a402ba20




Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

soniaAguilarPeiron committed:
Add pause/unpause bulk actions button in the alert list view

soniaAguilarPeiron committed:
add delete bulk action

soniaAguilarPeiron committed:
Add tracking and refactor FolderActionMenuItem

soniaAguilarPeiron committed:
Merge remote-tracking branch 'origin/main' into alerting/bulk-actions-part1

soniaAguilarPeiron committed:
update translations and text

soniaAguilarPeiron commented on public/app/features/alerting/unified/components/MoreButton.tsx:
```diff
@@ -4,18 +4,21 @@ import { Button, ButtonProps, Icon, Stack } from '@grafana/ui';
 
-const MoreButton = forwardRef(function MoreButton(props: ButtonProps, ref: Ref<HTMLButtonElement>) {
+const MoreButton = forwardRef(function MoreButton(
+  props: ButtonProps & { title?: string },
```
There was the option to create a new component exactly like this but with different title. I decided to reuse this one, but I can follow the other approach.

soniaAguilarPeiron committed:
update text

soniaAguilarPeiron committed:
add finally

soniaAguilarPeiron committed:
use ability for delete action

soniaAguilarPeiron committed:
don't show bulk actions if no action is allowed

soniaAguilarPeiron committed:
Merge remote-tracking branch 'origin/main' into alerting/bulk-actions-part1

soniaAguilarPeiron committed:
invalidate tags for delete action

soniaAguilarPeiron commented on public/app/features/alerting/unified/api/alertingFolderActionsApi.ts:
```diff
@@ -0,0 +1,61 @@
+          },
+        };
+      },
+      invalidatesTags: (_, _error, { namespace }) => {
```
I'm not sure about this invalidation 🤔

soniaAguilarPeiron committed:
revert invalidating and redirect to list page instead

soniaAguilarPeiron committed:
redirect when pausing/unpausing

soniaAguilarPeiron committed:
add pause/unpause endpoints

soniaAguilarPeiron committed:
add translations

soniaAguilarPeiron commented:
/deploy-to-hg

soniaAguilarPeiron committed:
add folder name in delete modal

soniaAguilarPeiron commented:
/deploy-to-hg

tomratcliffe commented:
Looks good to me but can't tell for sure until the API is actually in place 😁 
I left nitpick changes in https://github.com/grafana/grafana/pull/104693

soniaAguilarPeiron committed:
Merge remote-tracking branch 'origin/main' into alerting/bulk-actions-part1

gillesdemey committed:
Merge remote-tracking branch 'origin/main' into alerting/bulk-actions-part1

soniaAguilarPeiron commented:
> Looks good to me but can't tell for sure until the API is actually in place 😁 I left nitpick changes in #104693

Thanks Tom! I addressed some of your suggestions in my PR. I added a comment to the ones i don't think we should merge.

soniaAguilarPeiron committed:
address pr review

soniaAguilarPeiron committed:
Merge remote-tracking branch 'origin/main' into alerting/bulk-actions-part1

soniaAguilarPeiron commented:
/deploy-to-hg

soniaAguilarPeiron committed:
Merge remote-tracking branch 'origin/main' into alerting/bulk-actions-part1

soniaAguilarPeiron commented:
/deploy-to-hg

soniaAguilarPeiron committed:
disable pause/unpause

soniaAguilarPeiron committed:
update redirect

soniaAguilarPeiron committed:
Merge remote-tracking branch 'origin/main' into alerting/bulk-actions-part1

soniaAguilarPeiron commented on public/app/features/alerting/unified/components/folder-bulk-actions/FolderBulkActionsButton.tsx:
```diff
@@ -0,0 +1,94 @@
+
+export const FolderBulkActionsButton = ({ folderUID }: Props) => {
+  const [pauseSupported, pauseAllowed] = useFolderAbility(FolderAction.Pause);
+  const canPause = pauseSupported && pauseAllowed && false; // lets disable pause for now
```
we will disable pause/unpause actions for this PR, as we depend on the new endpoints to be finished.

soniaAguilarPeiron commented:
/deploy-to-hg

soniaAguilarPeiron committed:
Merge remote-tracking branch 'origin/main' into alerting/bulk-actions-part1

soniaAguilarPeiron commented:
/deploy-to-hg

soniaAguilarPeiron commented on public/app/features/alerting/unified/components/folder-bulk-actions/MenuItemPauseFolder.tsx:
```diff
@@ -0,0 +1,49 @@
+  /**
+   * Method invoked after the request to notify that the bulk action has been completed
+   */
+  onActionSucceed?: () => void;
```
maybe we could delete this method for now as is not used yet?

Proximyst committed:
Merge branch 'main' into alerting/bulk-actions-part1

Proximyst commented:
/deploy-to-hg

Proximyst commented:
/deploy-to-hg

Proximyst committed:
Merge branch 'main' into alerting/bulk-actions-part1

Proximyst commented:
/deploy-to-hg

soniaAguilarPeiron committed:
remove unnecessary prop

konrad147 commented on public/app/features/alerting/unified/components/folder-bulk-actions/FolderBulkActionsButton.tsx:
```diff
@@ -0,0 +1,94 @@
+    </>
+  );
+
+  return (
```
I have a slight concern regarding the screen real estate used by this button. If the expectation is that the Folder Actions are not used frequently, the current button size appears quite large. Could we consider alternatives that require less space

soniaAguilarPeiron replied:
I've just pushed a commit using the vertical elipsis icon instead: 
<img width="397" alt="Screenshot 2025-05-06 at 16 37 04" src="https://github.com/user-attachments/assets/01fdd87b-590e-4de7-9286-af65a78bc2b0" />

konrad147 commented on public/app/features/alerting/unified/hooks/useAbilities.ts:
```diff
@@ -91,6 +91,12 @@ export enum AlertRuleAction {
 }
 
+// this enum list all of the bulk actions we can perform on a folder
+export enum FolderAction {
```
Could we be more precise in naming here? How about `BulkFolderAction`
And then `PauseRules`, `DeleteRules`?
The current name suggests we try to pause or delete a folder, not rules in it

konrad147 commented on public/app/features/alerting/unified/hooks/useAbilities.ts:
```diff
@@ -113,10 +119,25 @@ export enum AlertingAction {
+/**
+ * This one will check for folder abilities
+ */
+export const useFolderAbilities = (): Abilities<FolderAction> => {
```
I do wonder if we should use `folderAbilities` name here. It conflicts a bit with folder abilities like renaming, deleting and it might be confusing.
How about having `useBulkActionAbilities` instead?

soniaAguilarPeiron replied:
I agree, but I would use something including the level of these bulk actions (folder): `useFolderBulkActionAbilities`

konrad147 commented on public/app/features/alerting/unified/components/folder-bulk-actions/FolderBulkActionsButton.tsx:
```diff
@@ -0,0 +1,94 @@
+        </>
+      )}
+      {canDelete && (
+        <Menu.Item
```
Shouldn't this also use `FolderActionMenuItem`?

soniaAguilarPeiron replied:
Nope, but I understand the confusion. I've just updated the `FolderActionMenuItem` name to `PauseUnpauseActionMenuItem`, as this component is specific for this pause/unpause actions.

konrad147 commented on public/app/features/alerting/unified/components/rules/RulesGroup.tsx:
```diff
@@ -217,6 +221,7 @@ export const RulesGroup = React.memo(({ group, namespace, expandAll, viewMode }:
             </div>
           </>
         )}
+        {isFolderBulkActionsEnabled && folderUID && isListView && <FolderBulkActionsButton folderUID={folderUID} />}
```
It looks like the current solution is mainly designed for removing imported rules. I'm wondering if we've considered adding this action to the Alert rules tab within the folders instead? This might not only limit the scope of changes required but also prevent the need for bulk actions on the alert list page, which could simplify future maintenance

soniaAguilarPeiron replied:
But thinking about it, I believe it makes more sense to place the bulk action in the alerting list view, as that’s where users naturally expect to manage alerting entities. Also, I don’t really see how this other approach improves maintainability.

soniaAguilarPeiron committed:
rename FolderActionMenuItem component to PauseUnpauseActionMenuItem

soniaAguilarPeiron committed:
address review comments

soniaAguilarPeiron committed:
fetch rules before redirecting in list view 1

soniaAguilarPeiron commented:
/deploy-to-hg

soniaAguilarPeiron committed:
Merge remote-tracking branch 'origin/main' into alerting/bulk-actions-part1

konrad147 reviewed (approved):
LGTM!
